### PR TITLE
GH#886 Solved privacy issue with comments

### PR DESCRIPTION
**Pull request by Sidsector9**
Private parents now have private children.

### CODE REVIEW

File: app/main/controllers/activity/RTMediaBuddyPressActivity.php [hunk 1]
```diff
@@ -847,4 +848,26 @@ public function rtm_bp_activity_permalink_access( $args ) {
 		return $has_access;
 
 	}
+
+	public function rtm_check_privacy_for_comments( $comment_id, $r, $activity ) {
```
Comment (milindmore22) [inline app/main/controllers/activity/RTMediaBuddyPressActivity.php]: the third argument seems unused `$activity `

Comment (milindmore22) [inline app/main/controllers/activity/RTMediaBuddyPressActivity.php]: Add Proper Documentation for function docblock

File: app/main/controllers/activity/RTMediaBuddyPressActivity.php [hunk 2]
```diff
@@ -847,4 +848,26 @@ public function rtm_bp_activity_permalink_access( $args ) {
...
+		$activity_id = $r['activity_id'];
+		$user_id = $r['user_id'];
+		$privacy_id = bp_activity_get_meta( $activity_id, 'rtmedia_privacy' );
+		$blog_id = get_current_blog_id();
+
+		if ( '60' === $privacy_id ) {
+			$wpdb->insert(
+				$db_prefix . $table_name,
+				array(
+					'activity_id' => $comment_id,
```
Comment (milindmore22) [inline app/main/controllers/activity/RTMediaBuddyPressActivity.php]: Add Proper alignment
```
'activity_id'        => $comment_id,
'user_id'		 => $user_id,
'privacy'		 => $privacy_id,
'blog_id'		 => $blog_id,
```

File: app/main/controllers/activity/RTMediaBuddyPressActivity.php [hunk 3]
```diff
@@ -847,4 +848,26 @@ public function rtm_bp_activity_permalink_access( $args ) {
 		return $has_access;
 
 	}
+
+	public function rtm_check_privacy_for_comments( $comment_id, $r, $activity ) {
+		global $wpdb;
+		$db_prefix = $wpdb->base_prefix;
```
Comment (milindmore22) [inline app/main/controllers/activity/RTMediaBuddyPressActivity.php]: Are we going to need base sites prefix? what if the site is multisite and using different prefix than the base site
may check this function `$wpdb->get_blog_prefix();`

Also please align all variable declarations properly

Comment (Sidsector9) [inline app/main/controllers/activity/RTMediaBuddyPressActivity.php]: @milindmore22 I did not know about this function, thank you!

**Comment by milindmore22** [comment]
@chandrapatel please do a final review

File: app/main/controllers/activity/RTMediaBuddyPressActivity.php [hunk 4]
```diff
@@ -847,4 +848,33 @@ public function rtm_bp_activity_permalink_access( $args ) {
...
+	/**
+	 * Makes the comments hidden (private) if the parent comment's
+	 * privacy is set to private.
+	 *
+	 * @param string $comment_id Activity id of the comment.
+	 * @param array  $r          Array of arguments.
+	 */
+	public function rtm_check_privacy_for_comments( $comment_id, $r ) {
+		global $wpdb;
+		$db_prefix = $wpdb->get_blog_prefix();
```
Comment (chandrapatel) [inline app/main/controllers/activity/RTMediaBuddyPressActivity.php]: Add blank line above

File: app/main/controllers/activity/RTMediaBuddyPressActivity.php [hunk 5]
```diff
@@ -847,4 +848,33 @@ public function rtm_bp_activity_permalink_access( $args ) {
...
+		$table_name = 'rt_rtm_activity';
+		$activity_id = $r['activity_id'];
+		$user_id = $r['user_id'];
+		$privacy_id = bp_activity_get_meta( $activity_id, 'rtmedia_privacy' );
+		$blog_id = get_current_blog_id();
```
Comment (chandrapatel) [inline app/main/controllers/activity/RTMediaBuddyPressActivity.php]: Align all this variables.

File: app/main/controllers/activity/RTMediaBuddyPressActivity.php [hunk 6]
```diff
@@ -847,4 +848,33 @@ public function rtm_bp_activity_permalink_access( $args ) {
...
 	}
+
+	/**
+	 * Makes the comments hidden (private) if the parent comment's
+	 * privacy is set to private.
+	 *
+	 * @param string $comment_id Activity id of the comment.
+	 * @param array  $r          Array of arguments.
+	 */
+	public function rtm_check_privacy_for_comments( $comment_id, $r ) {
```
Comment (chandrapatel) [inline app/main/controllers/activity/RTMediaBuddyPressActivity.php]: Add `! empty()` check for `$commentd_id` and `$r`. So it prevent any notices/warnings.

Comment (Sidsector9) [inline app/main/controllers/activity/RTMediaBuddyPressActivity.php]: But what if the `$comment_id` is `0`? In that case `empty()` will still return `true` which shouldn't be the case.

**Pull request merged**